### PR TITLE
OCR support tool

### PR DESCRIPTION
**Pull request by Jens-Co**
sudo apt install tesseract-ocr -y

main tessdata folder : /usr/share/tesseract-ocr/4.00/tessdata

Ill leave it as draft since there are some things still needed to be added.

### CODE REVIEW

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: src/main/java/org/geysermc/discordbot/storage/ServerSettings.java [hunk 1]
```diff
@@ -142,6 +142,23 @@ public static boolean shouldNotLogChannel(MessageChannel channel) {
         return dontLog.contains(channel.getId());
     }
 
+    /**
+     * Check if the given channel should be excluded from checking errors
+     *
+     * @param channel The {@link TextChannel} to check
```
Comment (Chew) [inline src/main/java/org/geysermc/discordbot/storage/ServerSettings.java:148]:
```suggestion
     * @param channel The {@link MessageChannel} to check
```

File: src/main/java/org/geysermc/discordbot/storage/ServerSettings.java [hunk 2]
```diff
@@ -142,6 +142,23 @@ public static boolean shouldNotLogChannel(MessageChannel channel) {
...
+     * @return If we should exclude the channel
+     */
+    public static boolean shouldNotCheckError(MessageChannel channel) {
+        Guild guild = getGuild(channel);
```
Comment (Chew) [inline src/main/java/org/geysermc/discordbot/storage/ServerSettings.java:152]: We don't need to define this if we just check if it's null, do `getGuild(channel) == null` for example

File: src/main/java/org/geysermc/discordbot/storage/ServerSettings.java [hunk 3]
```diff
@@ -142,6 +142,23 @@ public static boolean shouldNotLogChannel(MessageChannel channel) {
...
+
+        if (guild == null) {
+            return true;
+        }
+
+        List<String> dontLog = getList(channel.getIdLong(), "dont-checkerror");
```
Comment (Chew) [inline src/main/java/org/geysermc/discordbot/storage/ServerSettings.java:158]: Inline this call

File: src/main/java/org/geysermc/discordbot/listeners/ErrorAnalyzer.java [hunk 4]
```diff
@@ -211,7 +248,6 @@ private void handleLog(MessageReceivedEvent event, String logContent) {
             } else {
                 embedBuilder.setDescription("We don't currently have automated responses for the detected errors!");
             }
-
```
Comment (Chew) [inline src/main/java/org/geysermc/discordbot/listeners/ErrorAnalyzer.java:214]: Don't remove this newline :(

File: src/main/java/org/geysermc/discordbot/listeners/ErrorAnalyzer.java [hunk 5]
```diff
@@ -74,21 +79,48 @@ public void onMessageReceived(MessageReceivedEvent event) {
...
+                    BufferedImage bi = ImageIO.read(attachment.retrieveInputStream().get());
+                    Dimension newMaxSize = new Dimension(2000, 1400);
+                    BufferedImage resizedImg = Scalr.resize(bi, Scalr.Method.BALANCED,
+                            newMaxSize.width, newMaxSize.height);
+                    errorText = getPicture.doOCR(resizedImg);
+
+                    ErrorHandler(errorText, embedBuilder, event);
+
+                } catch (InterruptedException | ExecutionException | IOException | TesseractException e) {
+                    e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (Jens-Co) [reply src/main/java/org/geysermc/discordbot/listeners/ErrorAnalyzer.java]: Not sure if what I did is something you would like.

**Comment by Jens-Co** [comment]
To-Do: add a cooldown on image lookups.

**Review by RednedEpic — APPROVED** [review]

**Pull request merged**